floats.go

```diff
@@ -69,9 +69,8 @@ func AddScaledTo(dst, y []float64, alpha float64, s []float64) []float64 {
 	if len(dst) != len(s) || len(dst) != len(y) {
 		panic("floats: lengths of slices do not match")
 	}
-	for i, val := range s {
-		dst[i] = y[i] + alpha*val
-	}
+	copy(dst, y)
```

Comment:
There's no reason to have a copy here. I can change DaxpyUnitary to take a destination argument: `DaxpyUnitary(alpha, x, y, z)` where `z[i] = alpha*x[i] + y[i]`